validator/src/main.rs

```diff
@@ -392,6 +392,20 @@ fn hash_validator(hash: String) -> Result<(), String> {
         .map_err(|e| format!("{:?}", e))
 }
 
+/// Returns the default shred storage size (include both data and coding
+/// shreds) based on the validator config.
+fn default_fifo_shred_storage_size(vc: &ValidatorConfig) -> u64 {
+    // The average shred size is around 1228 bytes.
+    // Here we reserve a little bit more than the 1228-byte average shred size
+    // to give extra storage to prevent FIFO from purging data that have not
+    // yet being marked as obsoleted by LedgerCleanupService.
+    const RESERVED_BYTES_PER_SHRED: u64 = 1600;
+    match vc.max_ledger_shreds {
+        Some(max_ledger_shreds) => max_ledger_shreds * RESERVED_BYTES_PER_SHRED,
+        None => DEFAULT_ROCKS_FIFO_SHRED_STORAGE_SIZE_BYTES,
```

Comment:
I misspoke in another comment; `max_ledger_shreds` will not always be a `Some(...)`. However, when it is a `None`, we should not enforce any limit here and the ledger should be allowed to grow indefinitely.

A usecase for this is our warehouse nodes that collect an epoch of ledger data without purging. I'm not sure we would switch over the warehouse nodes to fifo; but regardless, we shouldn't allow for possibility of unintended behavior

Reply:
Interesting.  Can I know what is the current settings of our warehouse nodes?  Does their LedgerCleanupService also purge nothing because `--limit-ledger-size` is not set?

If the answer is yes, then warehouse nodes are already configured to grow indefinitely.  In that case, they are also safe to switch to FIFO if we allow the ledger to grow indefinitely when max_ledger_shreds is None.

Reply:
https://github.com/solana-labs/solana/blob/cdb95a362faf66e9384ece57054b45e805d0caa2/core/src/tvu.rs#L297-L306

Reply:
Got it!  So it should be unlimited for FIFO in that case.  Let me update the PR.